### PR TITLE
add regex support

### PR DESCRIPTION
**Pull request by tomSawkins**
#2

Added support for RegExp Objects with tests :+1: 

Still the speed demon it was as well.

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -53,6 +53,33 @@ describe('When cloning', function() {
...
+
+		beforeEach(function() {
+			expected = new RegExp("plow", "gi");
+			actual = clone(expected);
+		});
+
+		it('should return a different RexExp Object', function() {
+			expect(actual).to.not.equal(expected);
+		});
+
```
Comment (codeandcats) [inline test/test.js:69]: Would you mind adding:

```
it('should have the same pattern', function() {
    expect(actual.source).to.equal(expected.source);
});

it('should have the same flags', function() {
    expect(actual.global).to.equal(expected.global);
    expect(actual.ignoreCase).to.equal(expected.ignoreCase);
    expect(actual.multiline).to.equal(expected.multiline);
});
```

File: test/test.js [hunk 2]
```diff
@@ -143,6 +170,36 @@ describe('When cloning', function() {
...
 			});
 		});
+
+		describe('of Regex', function() {
+			var expected;
+			var actual;
+			var hisName = 'Mr Plow, thats his name, that name again is Mr Plow'
+
+			beforeEach(function() {
+				expected = [new RegExp("plow", "gi"), /mr/gi];
```
Comment (codeandcats) [inline test/test.js]: Could you add a regex with no flags

```
expected = [new RegExp("plow", "gi"), /mr/gi, /name/];
```

File: test/test.js [hunk 3]
```diff
@@ -143,6 +170,36 @@ describe('When cloning', function() {
...
+			it('should return array containing new RexExp instances', function() {
+				for (var i=0; i<expected.length; i++) {
+					expect(actual[i]).to.not.equal(expected[i]);
+				}				
+			});
+
+			it('should return array of matching RegExp', function() {
+				for (var i=0; i<expected.length; i++) {
+					expect(hisName.search(actual[i])).to.equal(hisName.search(expected[i]));
+
```
Comment (codeandcats) [inline test/test.js:210]: Mo tests

```
expect(actual[i].source).to.equal(expected[i].source);

expect(actual[i].global).to.equal(expected[i].global);
expect(actual[i].ignoreCase).to.equal(expected[i].ignoreCase);
expect(actual[i].multiline).to.equal(expected[i].multiline);
```

File: index.js [hunk 4]
```diff
@@ -27,6 +27,9 @@
 					result = new Date();
 					result.setTime(value.getTime());
 					return result;
+				} else if (value instanceof RegExp) {					
+					result = newRegExp(value);
+   					return result;
```
Comment (codeandcats) [inline index.js:33]: Minor thing, can you fix up this indentation pls?

**Comment by codeandcats** [comment]
Thanks man, nice work! I appreciate it. 👍 

If you can fix up these few things I'll merge it in. Cheers!

**Comment by tomSawkins** [comment]
Should be all fixed up, all tests passing like a charm

**Pull request merged**